StdMinMaxElement.md

```diff
@@ -0,0 +1,412 @@
+ - [`min_element`](#kokkosexperimentalmin_element)
+ - [`max_element`](#kokkosexperimentalmax_element)
+ - [`minmax_element`](#kokkosexperimentalminmax_element)
+
+
+------
+
+
+## `Kokkos::Experimental::min_element`
+
+```cpp
+//
+// overload set accepting iterators
+//
+template <class ExecutionSpace, class IteratorType>
+auto min_element(const ExecutionSpace& exespace,                        (1)
+                 IteratorType first, IteratorType last);
+
+template <class ExecutionSpace, class IteratorType>
+auto min_element(const std::string& label,                              (2)
+                 const ExecutionSpace& exespace,
+                 IteratorType first, IteratorType last);
+
+template <class ExecutionSpace, class IteratorType, class ComparatorType>
+auto min_element(const ExecutionSpace& exespace,                        (3)
+                 IteratorType first, IteratorType last,
+                 ComparatorType comp);
+
+template <class ExecutionSpace, class IteratorType, class ComparatorType>
+auto min_element(const std::string& label,                              (4)
+                 const ExecutionSpace& exespace,
+                 IteratorType first, IteratorType last,
+                 ComparatorType comp);
+
+//
+// overload set accepting views
+//
+template <class ExecutionSpace, class DataType, class... Properties>
+auto min_element(const ExecutionSpace& exespace,                        (5)
+                 const ::Kokkos::View<DataType, Properties...>& view);
+
+template <class ExecutionSpace, class DataType, class... Properties>
+auto min_element(const std::string& label,                              (6)
+                 const ExecutionSpace& exespace,
+                 const ::Kokkos::View<DataType, Properties...>& view);
+
+template <class ExecutionSpace, class DataType, class ComparatorType, class... Properties>
+auto min_element(const ExecutionSpace& exespace,                        (7)
+                 const ::Kokkos::View<DataType, Properties...>& view,
+                 ComparatorType comp);
+
+template <class ExecutionSpace, class DataType, class ComparatorType, class... Properties>
+auto min_element(const std::string& label,                              (8)
+                 const ExecutionSpace& exespace,
+                 const ::Kokkos::View<DataType, Properties...>& view,
+                 ComparatorType comp);
+```
+
+### Description
+
+- (1,2,5,6): Finds the smallest element in the range `[first, last)` (1,2)
+  or in `view` (5,6) using `operator<` to compare two elements
+
+- (3,4,7,8): Finds the smallest element in the range `[first, last)` (3,4)
+  or in `view` (7,8) using the binary functor `comp` to compare two elements
+
+### Parameters and Requirements
+
+- `exespace`:
+  - execution space instance
+- `label`:
+  - used to name the implementation kernels for debugging purposes
+  - for 1,3 the default string is: "Kokkos::min_element_iterator_api_default"
```

Comment:
I would consider removing the default strings description (everywhere) just for the sake of brevity. Maybe one general comment at the top would be enough? (that every algorithm gets a default value for `label` etc.)

Reply:
i am strongly for keeping these. I would say documentation is the only place where a bit more verbosity helps. 
I think it is useful to have labels here so that users don't have to go and look elsewhere.... in theory, we should have each algo to be as self-contained as possible. I think we are already pretty good in terms of balance.